lib/ecto/enum.ex

```diff
@@ -96,8 +99,16 @@ defmodule Ecto.Enum do
     on_load = Map.new(mappings, fn {key, val} -> {val, key} end)
     on_dump = Map.new(mappings)
     on_cast = Map.new(mappings, fn {key, _} -> {Atom.to_string(key), key} end)
-
-    %{on_load: on_load, on_dump: on_dump, on_cast: on_cast, mappings: mappings, type: type}
+    embed_as = if opts[:dump_embed?], do: :dump, else: :self
```

Comment:
What if we name the option `embed_as: :enum | :values`?

Reply:
I agree its better to have names than the boolean interface, but I would suggest other names that make what happens more obvious (specially since `Ecto.Enum` receive `:values` which are the "enumerated options")

Perhaps `embed_as: :values | :dumped_values`?

Reply:
Oh, that's what I meant to say but I got confused. 😅  Maybe `:values | :dumped`? Or maybe just stick with `:self | :dump`?

Reply:
I like `:values | :dumped`, lets go with it :)

Reply:
Pushed a commit changing this